Core/clim-basic/geometry/region-utilities.lisp

```diff
@@ -338,57 +329,87 @@
 
 ;;; -- Intersection Line/Polygon ---------------------------------------------
 
-(defun map-over-schnitt-gerade/polygon (fun x1 y1 x2 y2 points)
-  ;; This calles 'fun' with the "Geradenfunktionsparameter" of each
-  ;; intersection of the line (x1,y1),(x2,y2) and the polygon denoted
-  ;; by 'points' in a "sensible" way. --GB
-  (let ((n (length points)))
+;;; By "overcut" we mean a scalar computed from the intersection point
+;;; between a polygon segment and an unbounded line.
+(defun map-over-overcuts-line/polygon (fun x1 y1 x2 y2 points
+                                       &aux (n (length points)))
+  ;; FUN is called for some intersection points between a line going
+  ;; through the segment (x1 y1 x2 y2) and the polygon. Each of these
+  ;; points may be potentially a vertice of a segment which is the
+  ;; intersection of the polygon and the line. Function argument is a
+  ;; parameter indicating where on the segment S the intersection
+  ;; point is positioned (where 0d0 is [x1,y1] and 1d0 is [x2,y2]). If
+  ;; parameter is not (<= 0d0 param 1d0) position falls outside the
+  ;; segment (but is still on the line). -- jd 2019-10-10
+  (flet ((call-fun (point)
+           (let ((px (point-x point))
+                 (py (point-y point)))
```

Comment:
* Use `multiple-value-bind` with `point-position` to get both at once (and save three generic function calls).
* Declare `call-fun` inline. 
* Use `(let ((fun (alexandria:ensure-function fun))) BODY)` so `fun` is coerced only once and can then be assumed to be a function (rather than a function designator).

Reply:
fixed